File: resolver/src/main/java/org/contextmapper/sample/tlas/infrastructure/migration/DatabaseSeedHandler.java
package org.contextmapper.sample.tlas.infrastructure.migration;

import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;
import org.contextmapper.sample.tlas.application.TlaGroupsApplicationService;
import org.contextmapper.sample.tlas.domain.tla.TLAGroup;
import org.contextmapper.sample.tlas.domain.tla.ThreeLetterAbbreviation;
import org.springframework.stereotype.Component;

import java.util.function.Function;

import static org.contextmapper.sample.tlas.domain.tla.TLAGroup.COMMON_GROUP;

@Component
public class DatabaseSeedHandler implements Function<String, String> {

    private static Log logger = LogFactory.getLog(DatabaseSeedHandler.class);

    private TlaGroupsApplicationService service;

    public DatabaseSeedHandler(final TlaGroupsApplicationService service) {
        this.service = service;
    }

    @Override
    public String apply(String unused) {
        logger.info("DatabaseSeedHandler called");
        try {
            service.addTLAGroup(new TLAGroup.TLAGroupBuilder(COMMON_GROUP)
                    .withDescription("Common TLA group")
                    .withTLA(new ThreeLetterAbbreviation.TLABuilder("TLA")
                            .withMeaning("Three Letter Abbreviation")
                            .withAlternativeMeaning("Three Letter Acronym")
                            .build())
                    .build());

            service.addTLAGroup(new TLAGroup.TLAGroupBuilder("AppArch")
                    .withDescription("Application Architecture")
                    .withTLA(new ThreeLetterAbbreviation.TLABuilder("ADR")
                            .withMeaning("Architectural Decision Record")
                            .withLink("https://adr.github.io/")
                            .build())
                    .build());

            service.addTLAGroup(new TLAGroup.TLAGroupBuilder("DDD")
                    .withDescription("Domain-Driven Design")
                    .withTLA(new ThreeLetterAbbreviation.TLABuilder("OHS")
                            .withMeaning("Open Host Service")
                            .build())
                    .withTLA(new ThreeLetterAbbreviation.TLABuilder("PL")
                            .withMeaning("Published Language")
                            .build())
                    .withTLA(new ThreeLetterAbbreviation.TLABuilder("CF")
                            .withMeaning("Conformist")
                            .build())
                    .withTLA(new ThreeLetterAbbreviation.TLABuilder("SK")
                            .withMeaning("Shared Kernel")
                            .build())
                    .withTLA(new ThreeLetterAbbreviation.TLABuilder("ACL")
                            .withMeaning("Anticorruption Layer")
                            .build())
                    .build());
        } catch (Exception e) {
            logger.error("Internal error has happened", e);
            return e.getMessage();
        }
        return "ok";
    }
}

File: manager/src/TLAManager.Infrastructure/Migration/DatabaseSeedHandler.cs
using Amazon.Lambda.Core;
using Microsoft.Extensions.DependencyInjection;
using TLAManager.Domain;
using TLAManager.Infrastructure.WebApi;
using TLAManager.Services;

namespace TLAManager.Infrastructure.Migration;

public class DatabaseSeedHandler : FunctionBase
{
    public async Task<string> SeedAsync(string unused, ILambdaContext context)
    {
        using var scope = ServiceProvider.CreateScope();
        var service = scope.ServiceProvider.GetService<ITlaGroupsApplicationService>()!;

        try
        {
            await service.AddTlaGroupAsync(new TLAGroup(
                new ShortName("common"),
                "Common Tla group",
                [
                    new(
                        new ShortName("TLA"),
                        "Three Letter Abbreviation",
                        new List<string> { "Three Letter Acronym" },
                        null,
                        TLAStatus.Accepted
                    )
                ]
            ));

            await service.AddTlaGroupAsync(new TLAGroup(
                new ShortName("AppArch"),
                "Application Architecture",
                [
                    new(
                        new ShortName("ADR"),
                        "Architectural Decision Record",
                        new List<string>(),
                        "https://adr.github.io/",
                        TLAStatus.Accepted
                    )
                ]
            ));

            await service.AddTlaGroupAsync(new TLAGroup(
                new ShortName("DDD"),
                "Domain-Driven Design",
                [
                    new(
                        new ShortName("OHS"),
                        "Open Host Service",
                        new List<string>(),
                        null,
                        TLAStatus.Accepted
                    ),
                    new(
                        new ShortName("PL"),
                        "Published Language",
                        new List<string>(),
                        null,
                        TLAStatus.Accepted
                    ),
                    new(
                        new ShortName("CF"),
                        "Conformist",
                        new List<string>(),
                        null,
                        TLAStatus.Accepted
                    ),
                    new(
                        new ShortName("SK"),
                        "Shared Kernel",
                        new List<string>(),
                        null,
                        TLAStatus.Accepted
                    ),
                    new(
                        new ShortName("ACL"),
                        "Anticorruption Layer",
                        new List<string>(),
                        null,
                        TLAStatus.Accepted
                    )
                ]
            ));
        }
        catch (Exception e)
        {
            context.Logger.LogError(e, "Internal error has happened");
            return e.Message;
        }

        return "ok";
    }
}